lib/cassandra/protocol/v4.rb

```diff
@@ -213,119 +232,161 @@ def actual_decode(buffer, fields, size, code)
           end
         end
 
-        def decode_response(opcode, protocol_version, buffer, size, trace_id, custom_payload, warnings)
+        def decode_response(opcode, protocol_version, buffer, size, trace_id,
+                            custom_payload, warnings)
```

Comment:
Much like in v3, this method is largely just formatting changes, but I did add some 'else' clauses to some cases where they were missing. Noted below.